mantidproject/mantid
# Instrument viewer projection fix

**Description of work.**

Fixes the camera projection when changing views  (Z+, Z-, etc) in the Instrument Viewer. The original camera projection was set, but never updated when a different view was selected. In some cases, this would cause the instrument to be off the screen.

**To test:**



Load several instruments and open the Instrument Viewer. Change the axis view to all combinations to verify the view changes and the instrument is displayed properly.

Load an instrument that is offset, such as `/HFIR/CG2/IPTS-20794/nexus/CG2_18339.nxs.h5`. The default Z- view should display the entire instrument. Switching between other views should also work.





Version into `ornl-next`: #32346

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
